src/usr/share/gpgmailer/gpgmailer-daemon.py

```diff
@@ -342,27 +370,128 @@ def verify_signing_config(config):
+      the system path that should take on the following ownership and permissions.
+    uid: The system user ID that should own the directory.
+    gid: The system group ID that should be associated with the directory.
+    mode: The umask of the directory access permissions.
+    """
+    logger.info('Creating directory %s.', os.path.join(system_path, program_dirs))
+
+    path = system_path
+    for directory in program_dirs.strip('/').split('/'):
+        path = os.path.join(path, directory)
+        if not os.path.isdir(path):
+            # Will throw exception if file cannot be created.
+            os.makedirs(path, mode)
+        os.chown(path, uid, gid)
+        os.chmod(path, mode)
+
+
+def check_if_mounted_as_tmpfs(pathname):
+    """Checks if a directory is mounted as tmpfs.
+
+    pathname: The directory to check.
+    Returns true if the directory is mounted as tmpfs.  False otherwise.
+    """
+    return 'none on {0} type tmpfs'.format(pathname) in subprocess.check_output('mount')
+
+
+# TODO: Add a switch to not spool to a RAM disk. (issue 21)
+def create_spool_directories(program_uid, program_gid):
+    """Mounts the program spool directory as a ramdisk and creates the partial and outbox
+    subfolders. Exit if any part of this method fails.
+
+    program_uid: The system user ID that should own all the spool directories.
+    program_gid: The system group ID that should be assigned to all the spool directories.
+    """
+    logger.info('Creating spool directories.')
+
+    try:
+        create_directory(
+            SYSTEM_SPOOL_DIR, PROGRAM_NAME, program_uid, program_gid,
+            # drwx--x--- gpgmailer gpgmailer
+            stat.S_IRUSR | stat.S_IWUSR | stat.S_IXUSR | stat.S_IXGRP)
+    except Exception as exception:
+        logger.critical('Could not create program spool directory. %s: %s',
+                        type(exception).__name__, str(exception))
+        raise exception
+
+    spool_dir = os.path.join(SYSTEM_SPOOL_DIR, PROGRAM_NAME)
+    mounted_as_tmpfs = check_if_mounted_as_tmpfs(spool_dir)
+
+    # If directory is not mounted as tmpfs and there is something in the directory, fail to
+    #   start.
+    if os.listdir(spool_dir) != [] and not mounted_as_tmpfs:
+        raise InitializationException('Program spool directory is not empty and not mounted '
+                                      'as a ramdisk. Startup failed.')
+
+    # If the program spool directory is empty and not already mounted as tmpfs, mount it as
+    #   tmpfs.
+    if not mounted_as_tmpfs:
+        logger.info('Attempting to mount the program spool directory as a ramdisk.')
+        subprocess.call(['mount', '-t', 'tmpfs', '-o', 'size=25%', 'none', spool_dir])
+
+    if not check_if_mounted_as_tmpfs(spool_dir):
+        raise InitializationException(
+            'Program spool directory was not mounted as a ramdisk. Startup failed.')
+
+    try:
+        # TODO: File Permissions Alone Should Be Enough to Protect Files In
```

Comment:
Why is this capitalized like this?

Reply:
This is the title of the bug. I'll live this alone.

Reply:
**leave

Reply:
Huh?